src/main/java/mate/academy/spring/controller/MovieSessionController.java

```diff
@@ -0,0 +1,71 @@
+package mate.academy.spring.controller;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.spring.dto.MovieSessionRequestDto;
+import mate.academy.spring.dto.MovieSessionResponseDto;
+import mate.academy.spring.model.MovieSession;
+import mate.academy.spring.service.MovieSessionService;
+import mate.academy.spring.service.mapper.MovieSessionMapper;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/movie-sessions")
+public class MovieSessionController {
+    private MovieSessionService movieSessionService;
+    private MovieSessionMapper movieSessionMapper;
+
+    public MovieSessionController(MovieSessionService movieSessionService,
+                                  MovieSessionMapper movieSessionMapper) {
+        this.movieSessionService = movieSessionService;
+        this.movieSessionMapper = movieSessionMapper;
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public long createMovieSession(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        MovieSessionResponseDto movieSessionResponseDto = movieSessionMapper
+                .toDto(movieSessionService.add(movieSessionMapper.toModel(movieSessionRequestDto)));
+        return movieSessionResponseDto.getId();
+    }
+
+    @GetMapping("/available")
+    @ResponseStatus(HttpStatus.OK)
+    public List<MovieSessionResponseDto> findAllAvailableSessions(
+            @RequestParam @DateTimeFormat(pattern = "dd.MM.yyyy")
+                    Long movieId,
+                    LocalDate date) {
```

Comment:
I think, you weren't testing your program (
```suggestion
            @RequestParam @DateTimeFormat(pattern = "dd.MM.yyyy")
                    LocalDate date, @RequestParam Long movieId) {
```

Reply:
Everything works